tailscale/resource_dns_split_nameservers.go

```diff
@@ -59,47 +60,39 @@ func resourceSplitDNSNameserversRead(ctx context.Context, d *schema.ResourceData
 	return nil
 }
 
-func resourceSplitDNSNameserversCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	client := m.(*Clients).V1
+func resourceSplitDNSNameserversCreateOrUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*Clients).V2
 	nameserversSet := d.Get("nameservers").(*schema.Set)
 	domain := d.Get("domain").(string)
 
 	nameserversList := nameserversSet.List()
 
-	req := make(tailscale.SplitDnsRequest)
+	req := make(tsclient.SplitDNSRequest)
 	var nameservers []string
 	for _, nameserver := range nameserversList {
 		nameservers = append(nameservers, nameserver.(string))
 	}
 	req[domain] = nameservers
 
 	// Return value is not useful to us here, ignore.
-	if _, err := client.UpdateSplitDNS(ctx, req); err != nil {
-		return diagnosticsError(err, "Failed to set dns split nameservers")
+	if _, err := client.DNS().UpdateSplitDNS(ctx, req); err != nil {
+		return diagnosticsError(err, "Failed to set create dns split nameservers")
 	}
 
 	d.SetId(domain)
 	return resourceSplitDNSNameserversRead(ctx, d, m)
 }
 
-func resourceSplitDNSNameserversUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	if !d.HasChange("nameservers") {
```

Comment:
Because of [this](https://github.com/tailscale/terraform-provider-tailscale/pull/419/files#r1723858929), we no longer need to check for changes to nameservers.

Reply:
Hmm I'm not deeply familiar with the underlying API but I don't quite follow; `"nameservers"` doesn't have `ForceNew: true`, so it seems like this field could be updated without recreating the resource?

Reply:
There are only two fields, `domain` and `nameservers`. Now that `domain` is `ForceNew`, any change to `domain` means that we'll create a new resource. If we're in `resourceSplitDNSNameserversCreateOrUpdate` and `domain` didn't change, that by definition means that `nameservers` must have changed, so there's no need to check for that explicitly.